### ISSUE TITLE
iiab-hotspot-on fails if IIAB installed via WiFi (/etc/sysconfig/dhcpd missing, so dhcpd won't start)

### ISSUE THREAD

**Issue by holta** [iiab/iiab#655]
@jvonau this failure (iiab-hotspot-on does not work on WiFi installs) is due to RPi3 unfortunately being in Appliance mode after installing IIAB over WiFi.  So /etc/sysconfig/dhcpd is missing in WiFi installs (as needed by /etc/systemd/system/dhcpd.service) so dhcpd won't start :(

The short-term workaround was to insert a live Ethernet cable, then run ./iiab-network, to force it into Gateway mode (as visible by "grep mode /etc/iiab/iiab.ini") which also created the missing file /etc/sysconfig/dhcpd &mdash; and turned on the hotspot FWIW.

Others don't have this luxury of a live Ethernet cable nearby.

How do we fix WiFi installs, so they're actually possible?  Such that RPi folk can later run iiab-hotspot-off anytime to reconnect to hotel-style WiFi/Internet for content/updates spontaneously on-demand?  (And yes likewise insert Ethernet anytime, those that are so lucky as to find such during provisioning or in developing countries etc!)

_(My shallow comprehension is that all RPi's should be set to gateway by default from the get-go and All The Time &mdash; but in any case how do we fix this?)_

**Comment by holta** [comment]
IMPORTANT CLARIF: i had to manually run "systemctl enable hostapd" as well, in order to force the hotspot to actually work on subsequent boots.

_Please those who can tell me how to tune [iiab-hotspot-on](https://github.com/holta/iiab/blob/iiab-startup/roles/network/templates/network/iiab-hotspot-on) and [iiab-hotspot-off](https://github.com/holta/iiab/blob/iiab-startup/roles/network/templates/network/iiab-hotspot-off) so everyday implementers/operators can use these reliably!_

**Comment by holta** [comment]
Great News @jvonau & @georgejhunt:

After the workaround above (that we do need to fix!) [iiab-hotstpot-off](https://github.com/holta/iiab/blob/iiab-startup/roles/network/templates/network/iiab-hotspot-off) and [iiab-hotspot-on](https://github.com/holta/iiab/blob/iiab-startup/roles/network/templates/network/iiab-hotspot-on) do work on-demand &mdash; this is Critical in allowing me/others to type "iiab-hotspot-on" every morning right before working alongside teachers in remote schools &mdash; and then typing "iiab-hotspot-off" late afternoon or evening back at the hostel/cohousing, when I/we urgently need to update+revise the content, iterating the curricular mix forward per educators' realpolitik requirements.

In short this will now allow for *incredible progress* when working in countries where Ethernet just plain doesn't exist (for all practical purposes anyway).  Hence allowing people to take Internet-in-a-Box far more seriously, as it reflects locals' genuine needs increasingly more every single day.  _Authentic "Fieldback" instead of yet another "throw it over the transom" static/1st world dream..._

**Comment by holta** [comment]
Plugging in an ad hoc (in-field) live Ethernet cord to an RPi3 running IIAB should always get it online without bureaucacy, so operators/implementers can immediately install updates/content etc.

So close but no cigar!  On these same RPi3 / IIAB 6.5 (master) systems going to Mexico, an IP address is assigned (visible within the eth0 section when running "ip a" after Ethernet cord's inserted) but the infamous `Temporary failure in name resolution` error has returned.

Manual override "systemctl restart dhcpcd" worked to fix DNS (in the exceptional case when the remote/field operator actually has a keyboard/screen or similar & training!) when dhcpcd had failed as follows:
```
root@box:~# systemctl status dhcpcd
?? dhcpcd.service - dhcpcd on all interfaces
   Loaded: loaded (/lib/systemd/system/dhcpcd.service; enabled; vendor preset: e
   Active: active (running) since Tue 2018-01-16 11:52:38 CST; 2min 52s ago
  Process: 447 ExecStart=/usr/lib/dhcpcd5/dhcpcd -q -b (code=exited, status=0/SU
 Main PID: 469 (dhcpcd)
   CGroup: /system.slice/dhcpcd.service
           ??????469 /sbin/dhcpcd -q -b

Jan 16 11:54:29 box.lan dhcpcd[469]: if_addaddress6: Permission denied
Jan 16 11:54:29 box.lan dhcpcd[469]: eth0: soliciting an IPv6 router
Jan 16 11:54:30 box.lan dhcpcd[469]: eth0: rebinding lease of 192.168.0.132
Jan 16 11:54:30 box.lan dhcpcd[469]: eth0: NAK: from 192.168.0.1
Jan 16 11:54:30 box.lan dhcpcd[469]: eth0: soliciting a DHCP lease
Jan 16 11:54:31 box.lan dhcpcd[469]: eth0: offered 192.168.0.131 from 192.168.0.
Jan 16 11:54:31 box.lan dhcpcd[469]: eth0: probing address 192.168.0.131/24
Jan 16 11:54:36 box.lan dhcpcd[469]: eth0: leased 192.168.0.131 for infinity
Jan 16 11:54:36 box.lan dhcpcd[469]: eth0: adding route to 192.168.0.0/24
Jan 16 11:54:36 box.lan dhcpcd[469]: eth0: adding default route via 192.168.0.1
```
FYI /etc/dhcpcd.conf contains this single `denyinterfaces` line below, as a result of having run `iiab-hotspot-on`:
```
denyinterfaces  wlan0
```
_Let's press for a legit+methodical way to address these 2 on-ticket workarounds, so ad hoc Ethernet-to-Internet and ad hoc WiFi-to-Internet (via `iiab-hotspot-off`) become increasingly reliable for all field operators!_

**Comment by jvonau** [comment]
On Tue, Jan 16, 2018 at 10:38 AM, A Holt <notifications@github.com> wrote:

> @jvonau <https://github.com/jvonau> this failure (iiab-hotspot-on does
> not work on WiFi installs) is due to RPi3 unfortunately being in Appliance
> mode after installing IIAB over WiFi. So /etc/sysconfig/dhcpd is missing in
> WiFi installs (as needed by /etc/systemd/system/dhcpd.service) so dhcpd
> won't start :(
>

​No shit Sherlock, you need to have more than 2 devices to have gateway
mode auto-detected, thus enabling the needed services on br0. Your wish to
reserve the non-gateway wired device for future gateway use complicates
what was an easy distinction. Now that the workaround to force br0 to be
present when installing via wifi is disabled this does not surprise me at
all. See
https://github.com/iiab/iiab/blob/b6d20fdbd7ddc4934ddd439a2bb450d21c5e8d10/roles/network/tasks/main.yml#L22-28

**Comment by holta** [comment]
Thank you to @jvonau who explained during today's community/team call (http://minutes.iiab.io) that we should look into uncommenting all or part of:

https://github.com/iiab/iiab/blob/master/roles/network/tasks/main.yml#L22-L28

e.g. “iiab_lan_iface: br0” (etc)

In order to establish that eth0 should be reserved (by default anyway!) as a gateway on RPi3 &mdash; even when a live gateway is not detected there during installation!

As needed by most all implementers &mdash; who very frequently need to perform remote/rapid updates using home DSL routers & cablemodems (during the days just prior to and just after installation, e.g. when a non-trained operator is asked to plug the RPi3 into a live Ethernet cable at a nearby cybercafe etc).

**Comment by holta** [comment]
@deldesir: do you happen to know if this is fixed / sufficiently solid when installing over WiFi?

(I presume your recent installations of IIAB 6.5/master were over Ethernet.)

**Comment by holta** [comment]
Time to declare victory!

(Don't hesitate to re-open this ticket as nec, if similar networking problems recur...)
